Add tests that deal with applying number format specifiers on Symbols

domfarolino opened the pull request:
This PR is a follow-up to https://github.com/w3c/web-platform-tests/pull/9008 which now deals only with the application of the `%s` format specifier on Symbols.

This PR now includes tests for the application of `%i`, `%d`, and `%f` format specifiers on Symbols. The approval of this PR is pending on further discussion between implementations. At the time of creating, the spec and these tests mandate that Console implementations throw an Error when applying the aforementioned format specifiers on Symbols, though many browsers do not behave in this way currently.

<!-- Reviewable:start -->

<!-- Reviewable:end -->

domfarolino committed:
Add tests that deal with applying number format specifiers on Symbols

w3c-bots commented:
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/24241)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/340771709)

domenic reviewed (approved):
These tests LGTM and test the spec correctly, so I'm marking approved. But, we should probably delay merging until we have more clarity on the spec questions in https://github.com/whatwg/console/issues/113.

domfarolino commented:
Sounds good, I've added a comment there to discuss two possible routes we could take with that part of the spec (one throwing on Symbol evaluation, the other not).

domfarolino committed:
Manually test number format specifiers applied to Symbols

Adds manual tests that apply number format specifiers on Symbols
for each Console method that uses Formatter. This also updates a
previous test of the same nature to call console.groupEnd after
calls to group() and groupCollapsed() to clean up the console output
and prevent unnecessary nesting. See https://github.com/whatwg/console/issues/113.

domfarolino commented:
I've added tests to reflect the most recent spec PR following discussion in https://github.com/whatwg/console/issues/113. Since we've opted to not throw when applying %i/%d/%f format specifiers on Symbols, I've added a manual test instead of an an `-any.js` test, so testers can see the output `NaN` in the developer tools.

I'll merge the spec PR and, Domenic can review/merge this upon signing off when he's back!

domfarolino committed:
Whoops - forgot to delete

domfarolino commented on console/console-number-format-specifiers-symbol-manual.html:
```diff
@@ -0,0 +1,26 @@
+
+for (method of methods) {
+  console[method]("%i", Symbol.for("description"));
+  if (method == "group" || method == "groupCollapsed") console.groupEnd();
```
This isn't the prettiest thing ever, but I'm kinda OK with it unless someone wants me to change it.
